### PR TITLE
Adsk Contrib - Add 'deprecated' C++ mechanism & cleanup code

### PR DESCRIPTION
**Pull request by hodoulp**
Signed-off-by: Patrick Hodoul <Patrick.Hodoul@autodesk.com>

Based on the known limitations in RB-2.0, a C++ deprecated mechanism must now be in place to highlight methods/classes that should not be used anymore (e.g. `Config::parseColorSpaceFromString()` for now) starting with versions 2.x. At the same time, the Legacy GPU path bug (fixed by #1352) imposes the removal of a 'faulty' method from the public API asap to force a transition to the 'new' Legacy GPU path.

The pull request adds deprecated macro (code from OpenEXR) and removes all the code associated to the 'old' Legacy GPU Path.

### CODE REVIEW

File: include/OpenColorIO/OpenColorIO.h [hunk 1]
```diff
@@ -2976,27 +2977,17 @@ class OCIOEXPORT GpuShaderCreator
...
- *    // The three potential scenarios are:
+ *    // The two potential scenarios are:
  *    //
  *    //   1. Instantiate the legacy shader description.  The color processor
  *    //      is baked down to contain at most one 3D LUT and no 1D LUTs.
  *    //
  *    //      This is the v1 behavior and will remain part of OCIO v2
  *    //      for backward compatibility.
  *    //
- *    OCIO::GpuShaderDescRcPtr shaderDesc
```
Comment (michdolan) [inline include/OpenColorIO/OpenColorIO.h:2987]: Is this the intended change here? Currently this is removing the generic shader desc from docs and leaving legacy and custom only. Sould it be removing legacy and leaving gneric and custom instead?

And with this new implementation, the new legacy render path is just a generic shader desc with special optimization to replicate OCIO v1 behavior correct?

Comment (hodoulp) [reply include/OpenColorIO/OpenColorIO.h:2987]: The wrong description was preserved. The 'legacy' shader desc is now gone for v2.1.

> And with this new implementation, the new legacy render path is just a generic shader desc with special optimization to replicate OCIO v1 behavior correct?

Yes.

Comment (hodoulp) [reply include/OpenColorIO/OpenColorIO.h:2987]: Fixed

**Review by michdolan — APPROVED** [review]

**Comment by hodoulp** [comment]
Friendly reminder that the pull request could now be merged next Monday.

**Review by doug-walker — APPROVED** [review]

**Pull request merged**